modules/core/04-channel/keeper/timeout.go

```diff
@@ -84,13 +84,6 @@ func (k Keeper) TimeoutPacket(
 		return types.ErrNoOpMsg
 	}
 
-	if channel.State != types.OPEN {
```

Comment:
couldn't we also just extend this to allow timeouts on `OPEN` and `FLUSHING` channels?

Reply:
that was my initial thought pre-discussion. This isn't spec'ed out in any way (i.e the check for `State != types.OPEN` also isn't there), so when we talked about it I remember us coming to the conclusion that it could be removed. Happy to wait for a review from @colin-axner when he gets back

Reply:
Why would you want to restrict when packets can be timed out?

Reply:
In theory that makes sense, but in practice it is problematic. There's a potential state where chainA has a closed channel and chainB doesn't. Until someone calls ChanCloseConfirm, all timeouts are valid, but with a restriction, after calling close confirm, timeouts are no longer valid, but that's a very loose assumption and I don't think apps are built to handle channel closes like that

My preference would be to open an issue on the spec if there's a concern. I don't see why ibc-go should be opinionated in this matter

Reply:
There's also no check on TimeoutOnClose, so this restriction can be bypassed unless you added it there as well

Reply:
Sounds fair enough to me!

Reply:
I think the other point I'd make is that IBC only specifies optimistic sends as unsafe (potentially not recoverable). I think it's important to build the assumption that all sends can be reverted regardless of what occurs